test/core/derivatives/implementation/backpack.py

```diff
@@ -1,65 +1,108 @@
         self.store_forward_io()
         return self.problem.derivative.jac_t_mat_prod(
             self.problem.module, None, None, mat
         )
 
-    def weight_jac_t_mat_prod(self, mat, sum_batch):
+    def weight_jac_t_mat_prod(self, mat, sum_batch):  # noqa: D102
         self.store_forward_io()
         return self.problem.derivative.weight_jac_t_mat_prod(
             self.problem.module, None, None, mat, sum_batch=sum_batch
         )
 
-    def bias_jac_t_mat_prod(self, mat, sum_batch):
+    def bias_jac_t_mat_prod(self, mat, sum_batch):  # noqa: D102
         self.store_forward_io()
         return self.problem.derivative.bias_jac_t_mat_prod(
             self.problem.module, None, None, mat, sum_batch=sum_batch
         )
 
-    def weight_jac_mat_prod(self, mat):
+    def weight_jac_mat_prod(self, mat):  # noqa: D102
         self.store_forward_io()
         return self.problem.derivative.weight_jac_mat_prod(
             self.problem.module, None, None, mat
         )
 
-    def bias_jac_mat_prod(self, mat):
+    def bias_jac_mat_prod(self, mat):  # noqa: D102
         self.store_forward_io()
         return self.problem.derivative.bias_jac_mat_prod(
             self.problem.module, None, None, mat
         )
 
-    def ea_jac_t_mat_jac_prod(self, mat):
+    def bias_ih_l0_jac_t_mat_prod(self, mat, sum_batch):  # noqa: D102
+        self.store_forward_io()
+        return self.problem.derivative.bias_ih_l0_jac_t_mat_prod(
+            self.problem.module, None, None, mat, sum_batch=sum_batch
+        )
+
+    def bias_hh_l0_jac_t_mat_prod(self, mat, sum_batch):  # noqa: D102
+        self.store_forward_io()
+        return self.problem.derivative.bias_hh_l0_jac_t_mat_prod(
+            self.problem.module, None, None, mat, sum_batch=sum_batch
+        )
+
+    def weight_ih_l0_jac_t_mat_prod(self, mat, sum_batch):  # noqa: D102
+        self.store_forward_io()
+        return self.problem.derivative.weight_ih_l0_jac_t_mat_prod(
+            self.problem.module, None, None, mat, sum_batch=sum_batch
+        )
+
+    def weight_hh_l0_jac_t_mat_prod(self, mat, sum_batch):  # noqa: D102
+        self.store_forward_io()
+        return self.problem.derivative.weight_hh_l0_jac_t_mat_prod(
+            self.problem.module, None, None, mat, sum_batch=sum_batch
+        )
+
+    def ea_jac_t_mat_jac_prod(self, mat):  # noqa: D102
         self.store_forward_io()
         return self.problem.derivative.ea_jac_t_mat_jac_prod(
             self.problem.module, None, None, mat
         )
 
-    def sum_hessian(self):
+    def sum_hessian(self):  # noqa: D102
         self.store_forward_io()
         return self.problem.derivative.sum_hessian(self.problem.module, None, None)
 
-    def input_hessian_via_sqrt_hessian(self, mc_samples=None):
+    def input_hessian_via_sqrt_hessian(self, mc_samples=None) -> Tensor:
+        """Computes the input hessian.
+
+        Args:
+            mc_samples: number of samples. Defaults to None.
```

Comment:
`None` uses the exact Hessian. `int` uses an MC approximation with the specified number of samples.